packages/core/src/command/rate-limiting/ddb-store.ts

```diff
@@ -0,0 +1,323 @@
+        extra: {
+          key,
+          context: "dynamo-store.get",
+          error,
+        },
+      });
+      throw new MetriportError(msg, error, { method: "get", key });
+    }
+    return {
+      totalHits: entry.data.totalHits,
+      resetTime: entry.data.resetTime ? this.buildDate(entry.data.resetTime) : undefined,
+    };
+  }
+
+  /**
+   * Method to increment a client's hit counter.
+   *
+   * @param key {string} - The identifier for a client.
+   *
+   * @returns {IncrementResponse} - The number of hits and reset time for that client.
+   *
+   * @public
+   */
+  async increment(key: string): Promise<IncrementResponse> {
+    const { log } = out(`DynamoStore increment - key ${key}`);
+    const now = buildDayjs().unix();
+    const currentEntry = await this.get(key);
+    if (currentEntry && currentEntry.resetTime && currentEntry.resetTime.getTime() < now) {
+      await this.resetKey(key);
+    }
+    const item = await this.ddbUtils.update({
+      partition: this.prefixCountKey(key),
+      expression: "ADD totalHits :inc SET resetTime = if_not_exists(resetTime, :resetTime)",
+      expressionAttributesValues: {
+        ":inc": 1,
+        ":resetTime": buildDayjs().add(this.windowMs, "milliseconds").unix(),
+      },
+    });
+    const entry = rateLimitCountSchema.safeParse(item.Attributes);
+    if (!entry.success) {
+      const error = entry.error;
+      const msg = "Error parsing DynamoStore count entry";
+      log(`${msg} - error: ${errorToString(error)}`);
+      capture.error(msg, {
+        extra: {
+          key,
+          context: "dynamo-store.increment",
+          error,
+        },
+      });
+      throw new MetriportError(msg, error, { method: "increment", key });
+    }
+    if (!entry.data.resetTime) {
+      throw new MetriportError("resetTime not set correctly", undefined, {
+        method: "increment",
+        key,
+      });
+    }
+    return {
+      totalHits: entry.data.totalHits,
+      resetTime: this.buildDate(entry.data.resetTime),
+    };
+  }
+
+  /**
+   * Method to decrement a client's hit counter.
+   *
+   * @param key {string} - The identifier for a client.
+   *
+   * @public
+   */
+  async decrement(key: string): Promise<void> {
+    const { log } = out(`DynamoStore decrement - key ${key}`);
+    const item = await this.ddbUtils.update({
+      partition: this.prefixCountKey(key),
+      expression: "ADD totalHits :inc",
+      expressionAttributesValues: {
+        ":inc": -1,
+      },
+    });
+    const entry = rateLimitCountSchema.safeParse(item.Attributes);
+    if (!entry.success) {
+      const error = entry.error;
+      const msg = "Error parsing DynamoStore count entry";
+      log(`${msg} - error: ${errorToString(error)}`);
+      capture.error(msg, {
+        extra: {
+          key,
+          context: "dynamo-store.decrement",
+          error,
+        },
+      });
+      throw new MetriportError(msg, error, { method: "decrement", key });
+    }
```

Comment:
Why parsing/error handling here if we don't do anything else with it?

Reply:
Yea.. just a posterity sake thing, will remove.

Reply:
removed.